src/Transform/Transform.ts

```diff
@@ -0,0 +1,122 @@
+import {
+	compressSpaces
+} from '../util';
+import Point from '../Point';
+import {
+	ITransform
+} from './types';
+import Document from '../Document';
+import Translate from './Translate';
+import Rotate from './Rotate';
+import Scale from './Scale';
+import Matrix from './Matrix';
+import Skew from './Skew';
+import SkewX from './SkewX';
+import SkewY from './SkewY';
+
+export {
+	Translate,
+	Rotate,
+	Scale,
+	Matrix,
+	Skew,
+	SkewX,
+	SkewY
+};
+
+function parseTransforms(transform: string) {
+	return compressSpaces(transform)
+		.trim()
+		.replace(/\)([a-zA-Z])/g, ') $1')
+		.replace(/\)(\s?,\s?)/g, ') ')
+		.split(/\s(?=[a-z])/);
+}
+
+function parseTransform(transform: string) {
+
+	const [
+		type,
+		value
+	] = transform.split('(');
+
+	return [
+		type.trim(),
+		value.trim().replace(')', '')
+	];
+}
+
+export default class Transform {
+
+	static transformTypes = {
+		translate: Translate,
+		rotate:    Rotate,
+		scale:     Scale,
+		matrix:    Matrix,
+		skewX:     SkewX,
+		skewY:     SkewY
+	};
+
+	private readonly transforms: ITransform[] = [];
+
+	constructor(
+		private readonly document: Document,
+		transform: string
+	) {
+
+		const data = parseTransforms(transform);
+
+		data.forEach((transform) => {
+
+			if (transform === 'none') {
+				return;
+			}
+
+			const [
+				type,
```

Comment:
This is very internal part of code, adding this type here is pointless.

Reply:
yea, i might still go through and type more things after we merge, even if it's internal, because it will help w/ future refactors  - maybe not in this area but just knowing what things you can pass into a arg can be valid at times.